File: SeriesTracker.API/Contracts/CreateSeriesRequest.cs
using System.ComponentModel.DataAnnotations;

namespace SeriesTracker.API.Contracts
{
    /// <summary>
    ///  Запрос для создания записи о просмотре аниме пользователем.
    /// </summary>
    /// <param name="AnimeId">Идентификатор аниме (обязательное поле).</param>
    /// <param name="CategoryId">Идентификатор категории, к которой относится аниме (обязательное поле).</param>
    /// <param name="WatchedEpisode">Количество просмотренных эпизодов (обязательное поле).</param>
    /// <param name="IsFavorite">Признак, является ли аниме избранным (обязательное поле).</param>
    public record CreateSeriesRequest
    (
        [Required(ErrorMessage = "AnimeId is required.")] int AnimeId,
        [Required(ErrorMessage = "CategoryId is required.")] int CategoryId,
        [Required(ErrorMessage = "WatchedEpisode is required.")] int WatchedEpisode,
        [Required(ErrorMessage = "IsFavorite is required.")] bool IsFavorite
    );
}

File: client/app/Models/Requests/CreateSeriesRequest.ts
/**
 * @interface CreateSeriesRequest
 * @description Объект, представляющий запрос на добавление аниме в список пользователя.
 * @property {number} animeId Идентификатор аниме.
 * @property {number} categoryId Идентификатор категории.
 * @property {number} watchedEpisode Количество просмотренных эпизодов.
 * @property {boolean} isFavorite Признак того, что аниме добавлено в избранное.
 */
export interface CreateSeriesRequest {
    animeId: number;
    categoryId: number;
    watchedEpisode: number;
    isFavorite: boolean;
}
